platform/eventing/nats.go

```diff
@@ -22,17 +26,47 @@ func NewNatsPublisher(natserver string, natsubject string) (client.Client, error
 	return natsClient, nil
 }
 
-// NewNatsConsumer creates a new event consumer using nats.
-func NewNatsConsumer(natserver string, natsubject string) (client.Client, error) {
-	consumer, err := cloudeventsnats.NewConsumer(natserver, natsubject, cloudeventsnats.NatsOptions())
-	if err != nil {
-		return nil, fmt.Errorf("can't create nats consumer: %v", err)
-	}
+// Consumer describes how a consumer looks like
+type Consumer struct {
+	Scheduler ConsumerScheduler
+	Listener  Listener
+}
```

Comment:
since they are not going to be on the same runtime, scheduler and listener shouldn't be bound like that